build-logic-settings/configuration-cache-compatibility/src/main/kotlin/gradlebuild.configuration-cache-compatibility.settings.gradle.kts

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright 2025 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import org.gradle.api.internal.plugins.DslObject
+
+run {
```

Comment:
ℹ️  An artificial block is required to avoid capturing references to the script, because those are not allowed in the isolated lifecycle callback. Unfortunately, even "static" looking functions at the top level of the script are compiled to regular instance methods of the implicit class of the script.